docs/architecture/adr-021-restricted-block-size.md

```diff
@@ -0,0 +1,50 @@
+# ADR 021: Restricted and Configurable Block Size
+
+## Changelog
+
+- 2023-05-12: Initial
+
+## Status
+
+Proposed
+
+## Context
+
+Currently, block sizes are controlled by three values, the smallest of which will determine what the protocol considers to be a valid block size.
+
+- `MaxBlockBytes` is a hard coded constant to the maximum total size of the protobuf encoded block, and therefore acts as a cap for `MaxBytes`. Currently set to 100 MB.
```

Comment:
[question] where is `MaxBlockBytes` defined? I can't find it in
- https://github.com/search?q=repo%3Acelestiaorg%2Fcelestia-app+MaxBlockBytes&type=code
- https://github.com/search?q=repo%3Acelestiaorg%2Fcelestia-core+MaxBlockBytes&type=code
- https://github.com/search?q=repo%3Acometbft%2Fcometbft+MaxBlockBytes&type=code

It's not obvious that `MaxBytes` and `MaxBlockBytes` are related and one is governance modifiable. Maybe we consider adopting a similar naming strategy as `MaxSquareSize` and `GovMaxSquareSize` because those are clearly related. In other words should we consider preserving `MaxBlockBytes` and renaming `MaxBytes` => `GovMaxBlockBytes`? My hunch is that the diff it will introduce with cometbft is not worth the extra clarity in naming.

Reply:
yeah, the only reason that I didn't change them is because `MaxBytes` is a consensus param, which changing the name might break a few things in the sdk or explorers. We could rename the constant in core

and ahh whoops, the constant is [`MaxBlockSizeBytes`](https://github.com/celestiaorg/celestia-core/blob/1b24b4894042234e7daf68c7924e4f3647b46e30/types/params.go#L15), will update